optimizer/evaluator/evaluator_test.go

```diff
@@ -382,6 +384,30 @@ func (s *testEvaluatorSuite) TestConvert(c *C) {
 	}
 }
 
+func (s *testEvaluatorSuite) TestCall(c *C) {
+	// To test this case, we need to fake a wrong builtin.Funcs
+	builtin.Funcs = map[string]builtin.Func{
+		"date": {nil, 8, 8, false, false},
+	}
+	ctx := mock.NewContext()
+
+	// Test case for unknown function
+	expr := &ast.FuncCallExpr{
+		FnName: model.NewCIStr("unknown"),
+		Args:   []ast.ExprNode{},
+	}
+	_, err := Eval(ctx, expr)
+	c.Assert(err, NotNil)
```

Comment:
You can assert equal error, like 
`c.Assert(ErrorInvalidOperation.Equals(err), IsTrue)`